### PR TITLE
Add documentation for Google ads connector

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Add documentation for Google ads connector

### CODE REVIEW

File: en/docs/reference/connectors/google-ads-connector/google-ads-overview.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+The Google Ads Connector allows you to access the Google Ads API. This lets you create, update, and delete ad campaigns and ads.
+
+To see the Google Ads Connector, navigate to the [connector store](https://store.wso2.com) and search for "google".
+
+<img src="{{base_path}}/assets/img/integrate/connectors/google-ads-store.png" title="Google Ads Connector Store" width="200" alt="Google Ads Connector Store"/>
+
+## Compatibility
+
+| Connector Version | Supported product versions |
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-overview.md:13]:
```suggestion
| Connector version | Supported product versions |
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+    }
+    ```
+
+## Error codes related to Google Ads Connector
+
+The connector may encounter errors during operation execution. When an error occurs, the `ERROR_MESSAGE` property will contain detailed information about the error. You can handle these errors using a `Fault Sequence` in your integration. For more information, refer to [Using Fault Sequences]({{base_path}}/learn/examples/sequence-examples/using-fault-sequences/).
+
+| Error code | Description |
+| -------- | ------- |
+| 701001 | A general error has occured. |
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:723]:
```suggestion
| 701001 | A general error has occurred. |
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+| 701002 | An error has occured due to an invalid configuration. |
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:724]:
```suggestion
| 701002 | An error has occurred due to an invalid configuration. |
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+| 701003 | An error has occured in the access token generation flow. |
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:725]:
```suggestion
| 701003 | An error has occurred in the access token generation flow. |
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+
+## Error codes related to Google Ads Connector
+
+The connector may encounter errors during operation execution. When an error occurs, the `ERROR_MESSAGE` property will contain detailed information about the error. You can handle these errors using a `Fault Sequence` in your integration. For more information, refer to [Using Fault Sequences]({{base_path}}/learn/examples/sequence-examples/using-fault-sequences/).
+
+| Error code | Description |
+| -------- | ------- |
+| 701001 | A general error has occurred. |
+| 701002 | An error has occurred due to an invalid configuration. |
+| 701003 | An error has occurred in the access token generation flow. |
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:725]: Shall we reword the description of each error code as follows (for clarity): 
- General error
- Invalid configuration error
- Error in access token generation flow

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+    ```json
+    {
+        "customer_id": "123123123",
+        "query": "SELECT customer_client.client_customer FROM customer_client"
+    }
+    ```
+
+## Error codes related to Google Ads Connector
+
+The connector may encounter errors during operation execution. When an error occurs, the `ERROR_MESSAGE` property will contain detailed information about the error. You can handle these errors using a `Fault Sequence` in your integration. For more information, refer to [Using Fault Sequences]({{base_path}}/learn/examples/sequence-examples/using-fault-sequences/).
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:719]:
```suggestion
The connector may encounter errors during operation execution. When an error occurs, the `ERROR_MESSAGE` property will contain detailed information about the error. You can handle these errors using a `Fault Sequence` in your integration. For more information, see [Using Fault Sequences]({{base_path}}/learn/examples/sequence-examples/using-fault-sequences/).
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-overview.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# Google Ads Connector Overview
+
+Google Ads are customizable advertising units that allow businesses to reach their target audience on Google and its associated platforms. These ads can be tailored with various formats, targeting options, and budget settings.
+
+The Google Ads Connector allows you to access the Google Ads API. This lets you create, update, and delete ad campaigns and ads.
+
+To see the Google Ads Connector, navigate to the [connector store](https://store.wso2.com) and search for "google".
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-overview.md:7]:
```suggestion
To see the Google Ads Connector, navigate to the [WSO2 Connector Store](https://store.wso2.com) and search for "Google".
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 8]
```diff
@@ -0,0 +1,725 @@
...
+    The `init` operation initializes the connection to Google Ads API.
+    <table>
+        <tr>
+            <th>Parameter name</th>
+            <th>Description</th>
+            <th>Required</th>
+        </tr>
+        <tr>
+            <td><code>name</code></td>
+            <td>Unique name to identify the connection by.</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:19]:
```suggestion
            <td>Unique name to identify the connection.</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 9]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>loginCustomerId</code></td>
+            <td>The ID of the manager account. Required when using a manager account to access an customer account.</td>
+            <td>No</td>
+        </tr>
+        <tr>
+            <td><code>tokenEndpoint</code></td>
+            <td>The HTTP endpoint that can be uses to obtain an access token. The default value is `https://www.googleapis.com/oauth2/v3/token`.</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:54]:
```suggestion
            <td>The HTTP endpoint that can be used to obtain an access token. The default value is `https://www.googleapis.com/oauth2/v3/token`.</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 10]
```diff
@@ -0,0 +1,725 @@
...
+    The `createCustomerClient` operation creates a new client under the provided manager. The new client customer is returned.
+    <table>
+        <tr>
+            <th>Parameter name</th>
+            <th>Description</th>
+            <th>Required</th>
+        </tr>
+        <tr>
+            <td><code>customerId</code></td>
+            <td>The ID of the manager under whom client customer is being created. Type: string</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:87]: Shall we format the param types as well?

`Type: <code>string</code>`

There are multiple occurrences.

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 11]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>customerClient</code></td>
+            <td>The new client customer to create. The resource name on this customer will be ignored. Type: object [Customer](https://developers.google.com/google-ads/api/rest/reference/rest/v17/Customer)</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:92]: Shall we update the external links as below:

`For more information, go to [Customer](https://developers.google.com/google-ads/api/rest/reference/rest/v17/Customer).`

There are multiple occurrences.

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 12]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>developerToken</code></td>
+            <td>The developer token for the Google Ads connection.</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>loginCustomerId</code></td>
+            <td>The ID of the manager account. Required when using a manager account to access an customer account.</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:49]:
```suggestion
            <td>The ID of the manager account. Required when using a manager account to access a customer account.</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 10 above)
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:87]:
```suggestion
            <td>The ID of the manager under whom the client customer is being created. Type: string</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 13]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>customerClient</code></td>
+            <td>The new client customer to create. The resource name on this customer will be ignored. Type: object [Customer](https://developers.google.com/google-ads/api/rest/reference/rest/v17/Customer)</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>accessRole</code></td>
+            <td>The proposed role of user on the created client customer. Accessible only to customers on the allow-list. Type: enum [AccessRole](https://developers.google.com/google-ads/api/rest/reference/rest/v17/AccessRole)</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:97]:
```suggestion
            <td>The proposed role of the user on the created client customer. Accessible only to customers on the allow-list. Type: enum [AccessRole](https://developers.google.com/google-ads/api/rest/reference/rest/v17/AccessRole)</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 14]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>operations</code></td>
+            <td>The list of operations to perform on individual campaigns. Type: object [CampaignOperation](https://developers.google.com/google-ads/api/rest/reference/rest/v17/CampaignOperation)</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>partialFailure</code></td>
+            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. Default is false. Type: boolean</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:307]:
```suggestion
            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. The default value is <code>false</code>. Type: boolean</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 15]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>operations</code></td>
+            <td>The list of operations to perform on individual ad groups. Type: object [AdGroupOperation](https://developers.google.com/google-ads/api/rest/reference/rest/v17/AdGroupOperation)</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>partialFailure</code></td>
+            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. Default is false. Type: boolean</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:405]:
```suggestion
            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. The default value is <code>false</code>. Type: boolean</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 16]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>operations</code></td>
+            <td>The list of operations to perform on individual ads. Type: object [AdGroupAdOperation](https://developers.google.com/google-ads/api/rest/reference/rest/v17/AdGroupAdOperation)</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>partialFailure</code></td>
+            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. Default is false. Type: boolean</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:495]:
```suggestion
            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. The default value is <code>false</code>. Type: boolean</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 17]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>operations</code></td>
+            <td>The list of operations to perform on individual ads. Type: object [AdOperation](https://developers.google.com/google-ads/api/rest/reference/rest/v17/AdOperation)</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>partialFailure</code></td>
+            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. Default is false. Type: boolean</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:611]:
```suggestion
            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. The default value is <code>false</code>. Type: boolean</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 18]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>query</code></td>
+            <td>The query string. Type: string</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>pageToken</code></td>
+            <td>Token of the page to retrieve. If not specified, the first page of results will be returned. Use the value obtained fromnextPageTokenin the previous response in order to request the next page of results. Type: string</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:674]:
```suggestion
            <td>Token of the page to retrieve. If not specified, the first page of the results will be returned. Use the value obtained fromnextPageTokenin the previous response to request the next page of results. Type: string</td>
```

Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:674]: Please fix the words `fromnextPageTokenin`.

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 19]
```diff
@@ -0,0 +1,725 @@
...
+            <td>No</td>
+        </tr>
+        <tr>
+            <td><code>returnTotalResultsCount</code></td>
+            <td>If true, the total number of results that match the query ignoring the LIMIT clause will be included in the response. Default is false. Type: boolean</td>
+            <td>No</td>
+        </tr>
+        <tr>
+            <td><code>summaryRowSetting</code></td>
+            <td>Determines whether a summary row will be returned. By default, summary row is not returned. If requested, the summary row will be sent in a response by itself after all other query results are returned. Type: enum [SummaryRowSetting](https://developers.google.com/google-ads/api/rest/reference/rest/v17/SummaryRowSetting)</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:694]:
```suggestion
            <td>Determines whether a summary row will be returned. By default, a summary row is not returned. If requested, the summary row will be sent in a response by itself after all other query results are returned. Type: enum [SummaryRowSetting](https://developers.google.com/google-ads/api/rest/reference/rest/v17/SummaryRowSetting)</td>
```

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 20]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+    </table>
+
+    **Sample configuration**
+
+    ```xml
+    <googleAds.init>
+        <connectionType>googleAds</connectionType>
+        <name>GGOGLE_ADS_CONN</name>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md:64]: Is `GGOGLE` a typo in the documentation? There are multiple occurrences.

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 17 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [13cd302](https://github.com/wso2/docs-mi/pull/933/commits/13cd302a06eb72f432461d655f7426a4238075cb)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 16 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [13cd302](https://github.com/wso2/docs-mi/pull/933/commits/13cd302a06eb72f432461d655f7426a4238075cb)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 15 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [13cd302](https://github.com/wso2/docs-mi/pull/933/commits/13cd302a06eb72f432461d655f7426a4238075cb)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md [hunk 21]
```diff
@@ -0,0 +1,725 @@
...
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>operations</code></td>
+            <td>The list of operations to perform on individual campaign budgets. Type: object [CampaignBudgetOperation](https://developers.google.com/google-ads/api/rest/reference/rest/v17/CampaignBudgetOperation)</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>partialFailure</code></td>
+            <td>If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. Default is false. Type: boolean</td>
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [13cd302](https://github.com/wso2/docs-mi/pull/933/commits/13cd302a06eb72f432461d655f7426a4238075cb)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 20 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [70b7f97](https://github.com/wso2/docs-mi/pull/933/commits/70b7f970e86199fba8829404016b58ac512ab3a9)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 18 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [70b7f97](https://github.com/wso2/docs-mi/pull/933/commits/70b7f970e86199fba8829404016b58ac512ab3a9)

Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [70b7f97](https://github.com/wso2/docs-mi/pull/933/commits/70b7f970e86199fba8829404016b58ac512ab3a9)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 10 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [70b7f97](https://github.com/wso2/docs-mi/pull/933/commits/70b7f970e86199fba8829404016b58ac512ab3a9)

File: en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md (hunk 11 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/google-ads-connector/google-ads-configuration.md]: Fixed with [3d8b309](https://github.com/wso2/docs-mi/pull/933/commits/3d8b309f07637aea9f374fc5b18c0cf72322444f)

**Review by arunans23 — APPROVED** [review]

**Pull request merged**